examples/resources/hcp_vault_secrets_app_iam_binding/resource.tf

```diff
@@ -0,0 +1,14 @@
+resource "hcp_service_principal" "sp" {
+  name = "example-sp"
+}
+
+resource "hcp_vault_secrets_app" "example" {
+  app_name    = "example-app-name"
+  description = "My new app!"
+}
+
+resource "hcp_vault_secrets_app_iam_binding" "example" {
+  resource_name = "secrets/project/41d107a7-eea6-4b5e-8481-508ab29e2b07/app/example-app-name"
```

Comment:
Similarly here too consider taking the resource id or name from the example app above.

Reply:
is this blocked on us on the HVS side adding resource id/name as responses in our APIs and tf resource outputs?

Reply:
@pmmukh. Yes, it is. I created a ticket for this https://hashicorp.atlassian.net/browse/VAULT-26121

Reply:
Until HVS returns it can we construct the resource name in the provider?